<!DOCTYPE html>

<html>

  <head>

    <meta charset='utf-8'>

    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    <meta name="viewport" content="width=device-width*0.8, initial-scale=2, maximum-scale=2">

    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Copse">

    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen" />

    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen" />

    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print" />

 </head>

<body style="background-color:#efdaa5;">

<link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Copse">

<p></p>
<p></p>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/logo_design.svg" align="top" style="width:800px;height:400px;"></p>
</div>

&nbsp; &nbsp; <div style="width:950px; margin:0 auto;">
<a href="#Introduction"> <img src = "../articles/skeleton.svg" width = "120px" height="80px" alt=""> <span style="color:black;">
<font size="3"> <b> Introduction </b></font>
</span></a> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
<a href="#Demo"> <img src = "../articles/Bone4.svg"
width = "120px" height="80px" alt=""> <span style="color:black;">
<font size="3"> <b> Demo </b></font>
</a> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
<a href="#Tutorial"> <img src = "../articles/skull.svg"
width = "120px" height="80px" alt=""> <span style="color:black;">
<font size="3"> <b> Tutorial </b></font>
</a> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
<a href="#Extras"> <img src = "../articles/bone.svg"
width = "120px" height="80px" alt=""> <span style="color:black;">
<font size="3"> <b> Extras </b></font>
</a> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
</div>


<style>
  .bottom-three {
     margin-bottom: 1cm;
  }
</style>


<p class="bottom-three">
</p>
<p class="bottom-three">
</p>
<p class="bottom-three">
</p>

<p class="bottom-three">
</p>
<p class="bottom-three">
</p>
<p class="bottom-three">
</p>

<div style="width:300px; margin:0 auto;">
<a href="https://github.com/kkdey/aRchaic"> <img src = "../articles/GitHub-Mark-64px.png"
width = "80px" height="50px" alt=""> <span style="color:black;">
<font size="3"> <b> </b></font></a> &nbsp; &nbsp; &nbsp; &nbsp; &nbsp;
<a href="https://gitter.im/aRchaic-site/Lobby"> <img src = "../articles/gitter-chat.png"
width = "120px" height="30px" alt=""> <span style="color:black;">
<font size="3"> <b> </b></font>
</a>
</div>

<p class="bottom-three">
</p>

<div style="width:350px; margin:0 auto;">
  <iframe src="https://ghbtns.com/github-btn.html?user=kkdey&repo=aRchaic&type=star&count=true" frameborder="0" scrolling="0" width="100px" height="100px" align = "middle"></iframe>
  <iframe src="https://ghbtns.com/github-btn.html?user=kkdey&repo=aRchaic&type=watch&count=true&v=2" frameborder="0" scrolling="0" width="100px" height="100px" align = "middle"></iframe>
  <iframe src="https://ghbtns.com/github-btn.html?user=kkdey&repo=aRchaic&type=fork&count=true&v=2" frameborder="0" scrolling="0" width="100px" height="100px" align = "middle"></iframe>
</div>

<p class="bottom-three">
</p>
<p class="bottom-three">
</p>

<style>
  .margin {
     padding-bottom: 0.1cm;
     padding-left: 2cm;
     padding-right: 4cm;
     padding-top: 1 cm;
     color: black;

  }
</style>

<p class ="margin" id = "Introduction"><span style="color:brown;"> <font size="7"><color = "brown;"><strong> Introduction </strong></span></font></p>

<p class = "margin"><font size="4"> <strong>aRchaic</strong> is a R/python software package developed by <a href="https://kkdey.github.io/">Kushal K Dey</a> and <a href="https://halasadi.wordpress.com/">Hussein Al Asadi</a> in collaboration with <a href="http://stephenslab.uchicago.edu/">Matthew Stephens</a> and <a href="https://jnpopgen.org/">John Novembre</a>. aRchaic provides tools for visualization, clustering and classification of ancient and/or modern DNA samples based on features highlighting DNA damage patterns. </font> </p>

<p class = "margin"><font size="4"> We provide demos and tutorials on how to apply <strong>aRchaic</strong> below. A pdf version of the vignettes is available at the link below.</font></p>

<style>
  .list {
     padding-bottom: 0.1cm;
     padding-left: 3cm;
     padding-right: 4cm;
     padding-top: 1 cm;
     color: black;
  }
</style>


<!-- <ul class="list">
<li><font size="4"> Exploratory analysis and powerful visualization of mutational profiles from a BAM file (applicable for both ancient and modern BAM files). </font> </li>
<p></p>
  <li><font size="4"> Clustering and summary visualization of a pool of ancient and modern BAM files, highlighting differences in patterns of mutational features. </font></li>
<p></p>
  <li><font size="4"> Classification of ancient and modern BAM files based on mutational profiles. </font></li></ul> -->

<h3><span class="margin" style = "color:brown"><font size="5.5"><em>Requirements </em></font></span></h3>

<p><span class="margin"><font size="4"> To install and run <strong>aRchaic</strong>, the user needs to have the following (installation instructions provided in the links) </span></font></p>

<ul class="list"><style="list-style-type:circle">
<li><a href="https://git-scm.com/"> <font size="4"> Git </font> </a></li>
  <li><a href="https://www.python.org/"> <font size="4"> Python (Version 2 or 3) </font> </a></li>
  <li><a href="http://pysam.readthedocs.io/en/latest/installation.html"><font size="4"> pysam (>= 0.10.0) </a></li>
  <li><a href="https://pypi.python.org/pypi/pyfaidx"><font size="4"> pyfaidx (>= 0.4.8.1) </a></li>
  <li><a href="https://cran.r-project.org/"> <font size="4"> R (&gt;= 3.2.0) </font> </a></li>
  <li> <font size="4"> R libraries - <a href="https://cran.r-project.org/web/packages/gridBase/index.html"> gridBase </a>, <a href="http://ggplot2.org/"> ggplot2 </a>, <a href="https://github.com/kkdey/maptpx"> maptpx </a>, <a href="https://github.com/kkdey/CountClust"> CountClust </a>, <a href="https://github.com/kkdey/Logolas"> Logolas </a>, <a href="https://github.com/kkdey/classtpx"> classtpx </a> </font>
</li>
</ul>

<h3><span class="margin" style = "color:brown"><font size="5.5"><em>Installation </em></font></span></h3>

<p class = "margin"><font size="4"> aRchaic uses Python interface for data preprocessing, a demo of which is provided in the next section, and R/RStudio for the main statistical analysis. R can be downloaded from <a href="https://cran.r-project.org/"> CRAN </a>. To start R, Windows users can click on the R GUI that comes with the install. Linux/Mac users can also run R through terminal by typing </font></p>

<pre><code>
R
</code></pre>

<p><span class="margin"><font size="4"> We recommend users to use <a href=https://www.rstudio.com/RStudio> RStudio </a> instead, which is a well maintained GUI for R and is extremely user friendly.</span></font></p>

<p><span class="margin"><font size="4"> Once in R/RStudio, the user can install  the Github developmental version of the <strong> aRchaic </strong> package through devtools. </span></font></p>

<pre><code>
devtools::install_github("aRchaic")
</code></pre>

<p><span class="margin"><font size="4"> <strong> aRchaic </strong> automatically loads the packages mentioned above. The routine to manually install the above packages are the following </span></font></p>

<pre><code>

###  Github R packages

devtools::install_github("kkdey/maptpx")
devtools::install_github("kkdey/CountClust")
devtools::install_github("kkdey/classtpx")

###  CRAN R packages
install.packages("ggplot2")
install.packages("gridBase")

### Bioconductor package
source("https://bioconductor.org/biocLite.R")
biocLite("Logolas")
</code></pre>

<p class="bottom-three">
</p>

<h3><span class="margin" style = "color:brown"><font size="5.5"><em>Input Format </em></font></span></h3>

<p class = "margin"><font size="4"> In <strong>aRchaic</strong> takes a MutationFeatureFormat file as input, which is a comma separated
.csv file generated from a BAM file through pysam (details in the next subsection). Each row lists the features for a mutation, and the frequency of mutations with similar feature occurring in the BAM file. An example is provided below. </font> </p>


<pre style="color:blue;"><code style="color:blue;">
TG->AC,1,45,G,C,-,7
CG->AT,38,31,G,G,-,1
CC->TC,56,4,G,C,+,1
TG->AT,93,28,T,T,-,1
AC->TA,1,39,T,G,+,8
CA->CA,1,50,C,C,+,1
CA->GT,4,83,T,T,+,1
CA->GC,22,52,C,C,-,1
GG->AG,39,0,C,G,-,22
</code></pre>

<p class = "margin"><font size="4"> In the first row above, the mutational feature is TG → AC, 1, 45, C, C and the count is 7. To be more specific, the muation corresponds to a G →  A  mismatch, which is flanked by T on the 5’ end and C on the 3’ end (this is also referred to as mutation trio). The mismatch occurs 1 base from the start of the read and 45 bases from end of the read, where start and end are determined with respect to the reference genome to which the read is mapped. Furthermore, the read corresponding to the mutation is flanked by G at the 5’ strand-break and C at the 3’ strand-break. The "-" signifies that the mismatch lies on read mapped to the reverse strand. </font></p>

<h3><span class="margin" style = "color:brown"><font size="5.5"><em>Generating MutationFeatureFormat File </em></font></span></h3>

<p class = "margin"><font size="4"> The user can generate the MutationFeatureFormat file using the "generate_summary_files.py" file under "aRchaic/python/". The python script requires the BAM file (.bam) and the corresponding index file (.bam.bai), which is generated from the BAM file through samtools and the reference genome to which the reads in the BAM file are mapped to. </font></p>


<pre><code>
usage: generate_summary_bams.py [-h] -b BAM -f FASTA -o OUT [--add-chr]
                                [--use-tags]

optional arguments:
  -h, --help            show this help message and exit
  -b BAM, --bam BAM     bam file
  -f FASTA, --fasta FASTA
                        reference file
  -o OUT, --out OUT     out file
  --add-chr             add chr prefix?, you can find out by running samtools
                        idxstats <your bamfile> | head -1
  --use-tags            use the MD and NM tags in your bam file?

</code></pre>




<!-- <ul class="list">
<li><font size="4"> The substitution type (C → T , C → A, C → G, T → A, T → C, T → G) and the flanking bases to the left (5' end) and right (3' end).  </font> </li>
<p></p>
  <li><font size="4"> The position of the mutation from the ends of the read. </font></li>
<p></p>
  <li><font size="4"> Which strand - forward or reverse - does the mutation occur in. </font></li>
<p></p>
  <li><font size="4"> The base from the reference genome at the 5' strand break of the fragment corresponding to the read. </font></li>
  </ul>

<p><span class="margin"> <font size="5.5"> Graphical Representation </font> </span></p>

<p class = "margin"><font size="4"> We provide a rich visualization of a sample mutational profile of an ancient Sardinian sample as follows.</font> </p>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/aRchaic_plot.png" align="top" style="width:800px;height:500px;"></p>
</div>

<p class = "margin"><font size="4"> From the above plot, it is easy to see that C → T mutations are abundant in that sample. Also the mutation trend plot demonstrates that the mutations are mainly concentrated at the ends of the reads and mutational rate decays quickly as one moves inwards from the ends of the reads. We also see an enrichment of purines (A and G) at the 5' strand break. All these features are characteristic of ancient DNA samples and act as a validation check. </font></p>

<p><span class="margin"> <font size="5.5"> Clustering </font> </span></p>

<p class = "margin"><font size="4"> When provided with a pool of ancient and/or modern samples, one can use these mutational features to cluster the samples. For clustering, we use a Grade of Membership (GoM) model, similar to <a href="https://www.genetics.ucla.edu/software/admixture/">ADMIXTURE </a> and <a href="https://bioconductor.org/packages/release/bioc/html/CountClust.html">CountClust </a>. It represents each sample as having grades of membership in underlying clusters. A demo for how the clustering model works is provided below. </font></p>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/overview.png" align="top" style="width:800px;height:500px;"></p>
</div>

<p><span class="margin"> <font size="5.5"> Classification </font> </span></p>
<p class = "margin"><font size="4"> For the classification problem, we consider a pool of known ancient and modern samples and train a classifier (SVM and a classification GoM model) on these samples. Once trained, this classifier is then fitted on new test sample - control samples or contaminated samples - with the aim to detect what percentage of reads of the sample are ancient (or modern) </font></p>

<p class="bottom-three">
</p>f
<p class="bottom-three">
</p>
<p class="bottom-three">
</p> -->

<p class ="margin" id = "Demo"><span style="color:brown;"> <font size="7"><color = "brown;"><strong> Demo </strong></span></font></p>

<p class = "margin"><font size="4"> The primary function of the aRchaic package is <code>aRchaic_cluster()</code> which performs clustering of the samples using a Grade of membership model, assigning to each sample a membership in the different clusters.
</font></p>

<pre><code>
  aRchaic::aRchaic_cluster(folders, K,
                           labs, run_from = "start",
                           output_dir, save_plot = TRUE)

#' folders: names of the folders containing MutationFeatureFormat files to be pooled
#' K : the number of clusters to fit
#' labs: the labels of the samples (defaults to folder name where they belong)
#' run_from: if "start", will run data processing and clustering steps from scratch.
             if "gom", performs only the clustering step
             if "plot", performs just the visualization
#' output_dir: The output directory where to store the output files and model output
#' save_plot: if TRUE, it saves the visualizations
</pre></code>

<p class = "margin"><font size="4"> The function generates as output a <code>model.rda</code> file storing the clustering model results, in the <code> output_dir </code> directory specified by the user. If <code> save_plot = TRUE </code>, it generates a <code> structure.png </code>, a STRUCTURE plot representation of the grades of membership of the samples from the clustering model provided. Also for each cluster <code> k </code>, it generates a logo plot visualization <code> logo_clus_k.png </code> files to demonstrate the structure of clusters. </font></p>

<p class = "margin"><font size="4"> An example visualization pooling the STRUCTURE plot and logo plot visualizations from the <code> structure.png </code> and <code> logo_clus_k.png </code> files are depicted below.



<p class ="margin" id = "Tutorial"><span style="color:brown;"> <font size="7"><color = "brown;"><strong> Demo </strong></span></font></p>


<p class = "margin"><font size="4"> In this section we provide an example of how an user can perform exploratory analysis and visualize the mutational profile for a single BAM file using <strong>aRchaic</strong>. The example file we use here is available under <a> data/Skoglund/ </a>. </font></p>

<p class = "margin"><font size="4"> First we use Python interface and the pysam package to extract the mutation signature counts data for the sample BAM file. </font></p>


<style>
  .div1 {
     padding-bottom: 0.5cm;
     padding-left: 2cm;
     padding-right: 2cm;
     padding-top: 0.5cm;
     color: brown;
  }
</style>

<pre><code>
  dir <- "../data/Skoglund/";
  out <- aggregate_signature_counts(dir,
                                    breaks = c(-1, seq(1,20,1)),
                                    flanking_bases = 1,
                                    pattern = "Ajv59.hs37d5.fa.merged.q30.csv")
  head(data.frame(t(out)))
</code></pre>

<p class = "margin"><font size="4"> The output would look like </font></p>

<pre style="color:blue;"><code style="color:blue;">
                    Ajv59.hs37d5.fa.merged.q30
  AA->CA_-_A_A_1                           9
  AA->CA_-_A_A_2                           1
  AA->CA_-_A_A_3                           1
  AA->CA_-_A_A_4                           1
  AA->CA_-_A_A_9                           1
  AA->CA_-_A_A_12                          1
</code></pre>

<p class = "margin"><font size="4"> We can club the signatures such that we only retain the left strand break base and the minimum distance of the mutation from the ends of the reads.We also club the signatures that are mutations of G and A (purines) to that of C and T (pyrimidines) on the opposite strand (for example G → A is clubbed with C → T on the opposite strand). </font></p>

<pre><code>
  clubbed_data <- club_signature_counts(out, flanking_bases = 1)
</pre></code>

<pre style="color:blue;"><code style="color:blue;">
                Ajv59.hs37d5.fa.merged.q30
  AC->AA_-_A_1                          26
  AC->AA_-_A_10                          1
  AC->AA_-_A_11                          1
  AC->AA_-_A_12                          1
  AC->AA_-_A_13                          1
  AC->AA_-_A_14                          3
</code></pre>

<p class = "margin"><font size="4"> We normalize the clubbed signature counts data and then plot the <strong> aRchaic</strong> logo plot. </font></p>

<pre><code>
  clubbed_data_normed <- clubbed_data/sum(clubbed_data)
  damageLogo_five(t(clubbed_data_normed),
                  renyi_alpha=100, xaxis_fontsize=12,
                  y_fontsize=12, pop_names = "Skoglund aDNA")

</code></pre>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/skoglund_plot.png" align="top" style="width:700px;height:500px;"></p>
</div>

<p class = "margin"><font size="4">  In <strong> aRchaic </strong>, we
also provide option to plot the read length distribution for the BAM file.
We also provide read length distribution for all reads, reads with C → T damages and reads with C → T damages within 5 bases from the ends of reads.
</strong></font></p>

<pre><code>
  par(mar=c(2,2,2,2))
  read_length_distribution(dir,
                           pattern = "Ajv59.hs37d5.fa.merged.q30.csv",
                           end_break = 5,
                           plot_layout = c(1,1),
                           cex.main = 0.8,
                           cex_legend = 0.7)
</code></pre>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/readlength_plot.png" align="top" style="width:700px;height:500px;"></p>
</div>

<p class = "margin"><font size="4">  There is also an option to plot the mutation trends from the ends of the reads from 5' and 3' ends of the reads. See the code below </font></p>

<pre><code>
  color=c("red","blue","cornflowerblue","black","cyan","darkblue",
        "brown4","burlywood","darkgoldenrod1","darkgray","deepskyblue","darkkhaki",
        "firebrick","darkorchid","hotpink","green","magenta","yellow",
         "azure1","azure4");

  par(mfrow=c(1,2))
  pattern_plot_full(file="../data/Skoglund/Ajv59.hs37d5.fa.merged.q30.csv",
                    pattern = c("C->T", "C->G", "C->A", "T->G", "T->A", "T->C",
                                "G->A", "G->C", "G->T", "A_>G", "A->T", "A->C"),
                    plot_type="right",
                    sample_name = "Skoglund",
                    cols = color)

  pattern_plot_full(file="../data/Skoglund/Ajv59.hs37d5.fa.merged.q30.csv",
                    pattern = c("C->T", "C->G", "C->A", "T->G", "T->A", "T->C",
                                "G->A", "G->C", "G->T", "A_>G", "A->T", "A->C"),
                    plot_type="left",
                    sample_name = "Skoglund",
                    cols = color)
</code></pre>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/mutation_trend_plot.png" align="top" style="width:700px;height:400px;"></p>
</div>

<p class ="margin" id = "Tutorial"><span style="color:brown;"> <font size="7"><color = "brown;"><strong> Tutorial </strong></span></font></p>

<p class ="margin" id = "Extras"><span style="color:brown;"><font size="7"><strong> Extras </strong></span></font></p>


  </body>
</html>
