<html>

  <head>

    <meta charset='utf-8'>

    <meta http-equiv="X-UA-Compatible" content="chrome=1">

    <meta name="viewport" content="width=device-width*0.8, initial-scale=2, maximum-scale=2">

    <link href="stylesheets/bootstrap.2.2.2.min.css" rel="stylesheet">

    <link href="stylesheets/broman2.css" rel="stylesheet" type="text/css" media="all">

    <link href="stylesheets/kbroman.css" rel="stylesheet" type="text/css" media="all">

    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Copse">

    <link rel="stylesheet" type="text/css" href="stylesheets/stylesheet.css" media="screen" />

    <link rel="stylesheet" type="text/css" href="stylesheets/pygment_trac.css" media="screen" />

    <link rel="stylesheet" type="text/css" href="stylesheets/print.css" media="print" />

 </head>

<body style="background-color:#ffffff;">

<!-- <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Copse"> -->

<p></p>
<p></p>

<div class="navbar">
      <div class="navbar-inner">
        <div class="container-narrow">
          <ul class="nav">
          	  <li><a href="index.html"><img src="../articles/logo_design.svg" width=200px title = "click here for Home"></a></li>
              <li><a href="introduction.html"><img src="../articles/skeleton.svg" width=60px title = "click here for Introduction"></a></li>
              <li><a href="demo.html"><img src="../articles/Bone4.svg" width=60px title = "click here for Demo"></a></li>
              <li><a href="tutorial.html"><img src="../articles/skull.svg" width=60px title = "click here for Tutorial"></a></li>
              <li><a href="extras.html"><img src="../articles/bone.svg" width=60px title = "click here for Extras"></a></li>
          </ul>
        </div>
      </div>
</div>

<!-- <div style="width:800px; margin:0 auto;">
<p><img src="../articles/logo_design.svg" align="top" style="width:800px;height:400px;"></p>
</div> -->


<style>
  .bottom-three {
     margin-bottom: 2cm;
  }
</style>

<style>
  .margin {
     padding-bottom: 0.1cm;
     padding-left: 2cm;
     padding-right: 4cm;
     padding-top: 1 cm;
     color: black;

  }
</style>

<style>
  .list {
     padding-bottom: 0.1cm;
     padding-left: 3cm;
     padding-right: 4cm;
     padding-top: 1 cm;
     color: black;
  }
</style> 


<p class="bottom-three">
</p>
<p class="bottom-three">
</p>
<p class="bottom-three">
</p>

<p class="bottom-three">
</p>



<div class ="margin" id="tutorial"><span style="color:brown;"> <font size="5"><color = "brown;">
<h1>Tutorial</h1>
<ul>
<li> <h3> Data Processing </h3>
<ul> <font size="4">
<li><a href="#generate_mff" style="color:brown;" >Generate MFF file from BAM: an example - <codeline1>generate_mff()</codeline1></a></li>
<li><a href="#mff_features" style="color:brown;">Features of the MFF file </a></li>
<li><a href="#mff_management" style="color:brown;">MFF file management </a></li>
<li><a href="#mff_aggregation" style="color:brown;"><codeline1> aRchaic_pool() </codeline1> Generate aggregated signature counts data from one or more MFF files </codeline1>
</a></li>
</font></ul></li>
<li> <h3> Data Filtering </h3>
<ul> <font size="4">
<li><a href="#filter_by_mutation" style="color:brown;">mutations - <codeline1> filter_by_mutation() </codeline1></a></li>
<li><a href="#filter_by_mutation_flank" style="color:brown;">mutations + flanking bases - <codeline1> filter_by_mutation_flank() </codeline1></a></li>
<li><a href="#filter_by_mutation_pos" style="color:brown;">mutations + position on read - <codeline1> filter_by_mutation_pos() </codeline1></a></li>
<li><a href="#filter_by_mutation_flank_pos" style="color:brown;">mutations + flanking_bases + position on read - <codeline1> filter_by_mutation_flank_pos() </codeline1></a></li>
<li><a href="#filter_out_strand" style="color:brown;">strand and no strand filtering - <codeline1> filter_out_strand() </codeline1></a></li>
<li><a href="#filter_out_strand_break" style="color:brown;">strand break and no strand break composition filtering - <codeline1> filter_out_strand_break() </codeline1></a></li>
</font></ul></li>
<li> <h3> Exploration of a MFF file </h3>
<ul><font size="4">
<li><a href="#aRchaic_view" style="color:brown;"> <codeline1> aRchaic_view() </codeline1>: Logo plot representation of a MFF file</a></li>
<li><a href="#read_length_distribution" style="color:brown;"> <codeline1> read_length_distribution() </codeline1>: read length distribution of a MFF file (w and w/o damaged reads)</a></li>
<li><a href="#aRchaic_mutation_trend" style="color:brown;"> <codeline1> aRchaic_mutation_trend() </codeline1>: Mutation trends along reads for each MFF file </a></li>
<li><a href="#flanking_base_composition" style="color:brown;"> <codeline1> flanking_base_composition() </codeline1>: Composition of flanking bases to mutations for each MFF file</a></li>
<li><a href="#strand_break_composition" style="color:brown;"> <codeline1> strand_break_composition() </codeline1>: Strand break composition of all reads in a MFF file</a></li>
</font></ul></li>
<li> <h3> Clustering of MFF files </h3>
<ul><font size="4">
<li><a href="#aRchaic_cluster" style="color:brown;"> <codeline1> aRchaic_cluster() </codeline1>: GoM model clustering and visualization of MFF files in one or multiple folders </a></li>
<li><a href="#aRchaic_cluster_beta" style="color:brown;"> <codeline1> aRchaic_cluster_beta() </codeline1>: a beta version of above aRchaic_cluster which is adaptive to filtered signatures </a></li>
</font></ul></li>
<li> <h3> Classification of MFF files </h3>
<ul><font size="4">
<li><a href="#aRchaic_class" style="color:brown;"> <codeline1> aRchaic_class() </codeline1>: classification model on MFF files - using SVM and classtpx </a></li>
</font></ul></li>
<li> <h3> Dimension reduction and visualization of MFF file </h3>
<ul><font size="4">
<li><a href="#aRchaic_pca" style="color:brown;"> <codeline1> aRchaic_pca() </codeline1>: Principal component analysis of MFF files </a></li>
<li><a href="#aRchaic_tsne" style="color:brown;"> <codeline1> aRchaic_tsne() </codeline1>: Non linear embedding of MFF files using t-SNE </a></li>
<li><a href="#aRchaic_sfa" style="color:brown;"> <codeline1> aRchaic_sfa() </codeline1>: Sparse factor analysis of MFF files </a></li>
</font></ul></li>
</ul>
</color></font></div>

<h3><span class="margin" style = "color:brown" id = "generate_mff"><font size="6"> Generate MFF file from BAM: an example </font></span></h3>

<p class = "margin"><font size="5"> aRchaic takes MutationFeatureFormat (MFF) file(s) as input.
We demonstrate how we generate a MFF file from an example BAM file (example.bam) using <codeline1> generate_mff </codeline1> Python function. We run the following command through terminal in Linux/Mac (Windows users can use PuTTy interface). </font></p>

<pre><code> 
  generate_mff example.bam ref.fa  

</code></pre>

<p class = "margin"><font size="5"> Along the .bam file, the user need to provide the reference genome FASTA file (ref.fa) and also should have the (.bam.bai) in the same directory as the .bam file. </font></p>

<p class = "margin"><font size="5"> The code will take around 5 minutes to run and generate the MFF file <codeline1> example.csv </codeline1> as output. </font></p>

<h3><span class="margin" style = "color:brown" id = "mff_features"><font size="6"> Features of an MFF file </font></span></h3>

<p class = "margin"><font size="5"> The MFF file is a <codeline1> .csv </codeline1> file, which records the counts of different mutational features. The first few lines of <codeline1> example.csv </codeline1> are as follows. </font></p>

<pre style="color:blue;"><code style="color:blue;">
TG->AC,1,45,G,C,-,7
CG->AT,38,31,G,G,-,1
CC->TC,56,4,G,C,+,1
TG->AT,93,28,T,T,-,1
AC->TA,1,39,T,G,+,8
CA->CA,1,50,C,C,+,1
CA->GT,4,83,T,T,+,1
CA->GC,22,52,C,C,-,1
GG->AG,39,0,C,G,-,22

</code></pre>

<p class = "margin"><font size="5"> In the first row above, the mutational feature is TG → AC, 1, 45, C, C and the count is 7. To be more specific, the mutation corresponds to a G →  A  mismatch, which is flanked by T on the 5’ end and C on the 3’ end (this is also referred to as mutation trio). The mismatch occurs 1 base from the start of the read and 45 bases from end of the read, where start and end are the 5' and 3'ends with respect to the reference genome to which the read is mapped. Furthermore, it tells us that the read corresponding to the mutation is flanked by G at the 5’ strand-break and C at the 3’ strand-break. The "-" signifies that the mismatch lies on read mapped to the reverse strand. </font></p>

<h3><span class="margin" style = "color:brown" id = "mff_management"><font size="6"> Management  of MFF files </font></span></h3>

<p class = "margin"><font size="5"> aRchaic is primarily aimed at clustering and comparative analysis of ancient/modern samples from different sources. Keeping in mind the possibility that the user may be re-using the same batch of data (from a single source or library prep) multiple times for different comparisons, we have tried to streamline our functions to ensure minimization of repeated tasks. </font></p>

<p class = "margin"><font size="5"> For example, signature aggregation from MFF file, which we discuss next, is a slightly time consuming step, taking around 15-30 seconds per MFF file. <codeline1> aRchaic_cluster () </codeline1> and <codeline1> aRchaic_pool() </codeline1> both save the results from signature aggregation on first pass that it can refer back to in subsequent passes. Many other examples of this type of automation in subsequent sections as we discuss the functions. </font></p>

<p class = "margin"><font size="5"> This automation, while facilitating reproducibity and re-usability, also calls for good management of the MFF files. To get the maximum efficiency of aRchaic codes,we suggest the user maintain a separate directory for all data from same library prep or lab that are likely to be used together. For example, we have made separate folders for <codeline1> "../data/Skoglund" </codeline1>,  <codeline1> "../data/Siberia" </codeline1> and <codeline1> "../data/Allentoft" </codeline1>, containing MFF files for  11, 1 and 150 samples. </font></p>


<h3><span class="margin" style = "color:brown" id = "aRchaic_view"><font size="6"> aRchaic_view (&nbsp) </font> : <font size="5"> Viewing a MFF file </font></span></h3>

<p class = "margin"><font size="5"> The user can display the mutational features in a MutationFeatureFormat (MFF) file by a logo plot using the <codeline1> aRchaic_view() </codeline1> function.</font></p>

<pre><code>
  aRchaic_view (file = "../data/Skoglund/Ajv59.hs37d5.fa.merged.q30.csv",
                breaks = c(-1, seq(1,20,1)),
                flanking_bases =1,
                logo.control = list(),
                title = "Skoglund aDNA",
                output_dir = NULL)
</code></pre>

<p class = "margin"><font size="5">  We supply as input arguments the name of the MFF file in <codeline1>file</codeline1>. The output will be a (.png) file of the logo plot of the MFF file, saved in <codeline1> output_dir </codeline1>. If <codeline1> output_dir </codeline1> is NULL, the plot will be saved in the current working directory. </font></p>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/skoglund_plot.png" align="top" style="width:700px;height:500px;"></p>
</div>

<p class = "margin"><font size="5"> The function provides control parameters that the user can toggle with. <codeline1>breaks</codeline1>represents how to bin the locations of the mutations on the read from each end. (see <codeline1> aggregate_signature_counts() </codeline1> documentation above for more details on the binning). <codeline1>flanking_bases </codeline1> represents the number of bases to the left and right of the mutation we record in our feature. The default is 1. <codeline1>title</codeline1> represents the title of the logo plot representation. <codeline1>logo.control()</codeline1> allows the user to play with the settings of the logo plot.</font></p>


<!-- <p class = "margin"><font size="5"> <codeline1>breaks</codeline1>represents how to bin the locations of the mutations on the read from each end. <codeline1>breaks = c(-1, seq(1,20,1)) </codeline1> means that only mutations occurring inside of 20 bases from ends of the reads are considered and each position is assigned to a bin of its own. <codeline1>breaks = c(-1, seq(1,20,2)) </codeline1> will pool mutations in consecutive bases in a single bin- for example, mutations in places 1 and 2 from the ends of the read will be assigned to bin 1 and so on. <codeline1>flanking_bases </codeline1> represents the number of bases to the left and right of the mutation we record in our feature. The default is 1, which means, we only record the base to the immediate left and right of the mutation (this is also called a mutation trio). </font></p> -->


<h3><span class="margin" style = "color:brown" id = "read_length_distribution"><font size="5.5"><em>read_length_distribution(): Read length distribution of each MFF file</em></font></span></h3>

<p class = "margin"><font size="5"> This function provides an estimated read length distribution of all reads in a MFF file. Every mutation in a MFF file comes from a read, for which we record the distance from the ends of the read. This provides us with the length of the read by summing the distance from the two ends. </font</p>

<p class = "margin"><font size="5">  <codeline1> read_length_distribution() </codeline1> provides density plot of read length distribution for reads containing any mutation, just C -> T mutations and the reads that contain a C->T mutation inside a small distance, denoted here by <codeline1> end_break </codeline1>, from one end of trhe read. Note here that C->T mutations at the ends of the read are a feature of ancient reads, so the latter gives an idea of the read length for the ancient reads. This is an exploratory scheme that can help us compare read lengths of ancient and modern reads in a contaminated sample.</font></p>

<p class = "margin"><font size="5"> We present an example of applying this function on a single file of the directory <codeline1> dir </codeline1>, here denoted by <codeline1> pattern </codeline1>. The user can add a vector of file names under <codeline1> pattern </codeline1> if he wants to see the read length distribution for multiple samples. </font></p>

<pre><code>
  read_length_distribution(dir = "../data/Skoglund/",
                         pattern = "Ajv59.hs37d5.fa.merged.q30.csv",
                         end_break = 5,
                         plot_layout = c(1,1),
                         cols = c("red", "green", "blue"),
                         cex_legend = 0.5,
                         cex.main = 0.5)
                         
</code></pre>

<p class = "margin"><font size="5"> The output from this function would look like the following </font></p>

<div style="width:800px; margin:0 auto;">
<p><img src="../articles/readlength_plot.png" align="top" style="width:700px;height:500px;"></p>
</div>

<p class = "margin"><font size="5"> We provide plot options in the form of <codeline1> plot_layout </codeline1> to fix the plotting window design, <codeline1> cols </codeline1> vector of colors for coloring the density plots for the three cases (reads with all mutations, with C->T mutations, with C->T mutations inside <codeline1> end_break </codeline1> from the end). The size of the legend and title are fixed by <codeline1> cex.legend </codeline1> and <codeline1> cex.main </codeline1>.</font</p>

<p class = "margin"><font size="5"> If <codeline1> pattern = NULL </codeline1>, then the function is run on all MFF files in the folder and the plots are saved in the <codeline1> plot_layout </codeline1> layout. An example is given below. </font></p>

<pre><code>
  read_length_distribution(dir = "../data/Allentoft/",
                         pattern = NULL,
                         end_break = 5,
                         plot_layout = c(3,3),
                         cols = c("red", "green", "blue"),
                         cex_legend = 0.5,
                         cex.main = 0.5)

</code></pre>



<p class ="margin" id = "Extras"><span style="color:brown;"><font size="7"><strong> Extras </strong></span></font></p>


  </body>
</html>
